### PR TITLE
Scenes / DashboardsLoader: Add variables migration

### PR DESCRIPTION
**Pull request by dprokop**
Adds VariableModel -> SceneVariable migration for currently supported variable types (custom, datasource, query).

### CODE REVIEW

File: public/app/features/scenes/dashboard/DashboardsLoader.ts [hunk 1]
```diff
@@ -136,6 +159,76 @@ export class DashboardLoader extends StateManagerBase<DashboardLoaderState> {
...
+          }
+        })
+        // TODO: Remove filter
+        // Added temporarily to allow skipping non-compatible variables
+        .filter((v): v is SceneVariable => Boolean(v))
+    );
+  }
+}
+
+export function createVariableFromLegacyModel(variable: VariableModel): SceneVariable {
```
Comment (ivanortegaalba) [inline public/app/features/scenes/dashboard/DashboardsLoader.ts:181]: If I'm not wrong, we also support [ConstantVariable](https://github.com/grafana/grafana/blob/main/public/app/features/scenes/variables/variants/ConstantVariable.ts).

File: public/app/features/scenes/dashboard/DashboardsLoader.test.ts [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+        id: 'query0',
+        global: false,
+        index: 0,
+        state: 'Done',
+        error: null,
+        description: null,
+        allValue: null,
+      };
+
+      const migrated = createVariableFromLegacyModel(variable);
```
Comment (ivanortegaalba) [inline public/app/features/scenes/dashboard/DashboardsLoader.test.ts:57]: If we're testing here, `DashboardLoader`. Shouldn't we test the migration through it? 

As I understand how we're doing our unit tests, the unit/subject of the test is the exported unit from the module. In the case of a scene object, we test the scene object and its integration with the scene. In this case, if we're testing the loader, I'd expect to initialize the loader with a template using the legacy model.

I also understand the DashboardLoader as it is, is hard to test. So I'm happy with this as soon as this serves as a signal for the DashboardLoader to be refactored in a way we can test this easily. WDYT?

**Review by ivanortegaalba — APPROVED** [review]
I did some manual testing and all variables in all scenarios look great! 

The only small issue is the constant variables disappear even they are supported. But it is not a bug, is a matter of enabling it.

File: public/app/features/scenes/dashboard/DashboardsLoader.ts (hunk 1 above)
Comment (dprokop) [reply public/app/features/scenes/dashboard/DashboardsLoader.ts:187]: Pushed a change with that just now

File: public/app/features/scenes/dashboard/DashboardsLoader.test.ts (hunk 2 above)
Comment (dprokop) [reply public/app/features/scenes/dashboard/DashboardsLoader.test.ts:57]: Yes, DashboardsLoader needs a refactor, then we can add a more verbose, integration test.

**Review by torkelo — APPROVED** [review]
Looks great!

**Pull request merged**